src/main/java/uk/gov/hmcts/probate/controller/DocumentController.java

```diff
@@ -450,4 +454,61 @@ public ResponseEntity<WillLodgementCallbackResponse> permanentlyDeleteRemovedWil
         documentGeneratorService.permanentlyDeleteRemovedDocumentsForWillLodgement(callbackRequest);
         return ResponseEntity.ok(willLodgementCallbackResponseTransformer.transform(callbackRequest));
     }
+
+    @PostMapping(path = "/validateAmendLegalStatement", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<CallbackResponse> validateAmendLegalStatement(
+            @RequestBody final CallbackRequest callbackRequest) {
+        final CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        final long caseId = caseDetails.getId();
+
+        log.info("Validating amend legal statement for case: {}", caseId);
+
+        final DocumentLink amendedLegalStatement = caseDetails.getData().getAmendedLegalStatement();
+
+        final Optional<String> validationErr = documentValidation.validateUploadedDocumentIsType(
+                caseId,
+                amendedLegalStatement,
+                MediaType.APPLICATION_PDF);
+
+        if (validationErr.isPresent()) {
+            final String validationMsg = validationErr.get();
+            log.info("case {} validation error: {}", caseId, validationMsg);
+            CallbackResponse err = CallbackResponse.builder().errors(List.of(validationMsg)).build();
+            return ResponseEntity.ok(err);
+        }
+
+        Optional<UserInfo> caseworkerInfo = userInfoService.getCaseworkerInfo();
+        CallbackResponse response = callbackResponseTransformer.transformCase(callbackRequest, caseworkerInfo);
+        return ResponseEntity.ok(response);
+    }
+
+    @PostMapping(path = "/amendLegalStatement", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<CallbackResponse> amendLegalStatement(
+            @RequestBody final CallbackRequest callbackRequest) {
+        final long caseId = callbackRequest.getCaseDetails().getId();
+        log.info("Amending legal statement for case: {}", caseId);
+
+        final CaseData caseData = callbackRequest.getCaseDetails().getData();
+        final ApplicationType applicationType = caseData.getApplicationType();
+        final DocumentLink amendedLegalStatement = caseData.getAmendedLegalStatement();
+
+        final String baseFileName = switch (applicationType) {
+            case PERSONAL -> "amendedLegalStatement";
+            case SOLICITOR -> "amendedLegalStatementGrantOfProbate";
```

Comment:
For solicitor case, the legal statement file name is based on the case type
i.e. legalStatementAdmon, legalStatementGrantOfProbate, legalStatementIntestacy
The amended one should follow this convention

Reply:
I've updated to use the method from the DocGenService here